src/kOS/Function/FunctionBase.cs

```diff
@@ -35,7 +35,7 @@ public object ReturnValue
                 internalReturn = value;
             }
         }
-        private object internalReturn = null;
+        private object internalReturn = 0; // really should be 'null', but kerboscript can't deal with that.
```

Comment:
I agree that they should match.  As to the conflation between a real value of 0 and a dummy value of 0, remember that these are only used in cases where the function is essentially void anyway, so there shouldn't be such a clash.  i.e. you'd have to be doing something like `set x to mylist:clear().` instead of just what you're supposed to do with is just `mylist:clear().` to be getting that return value.  So it's only happening in cases where the docs don't mention any intended return value you should be looking for.  There can't be a clash between "it returned meaningful zero" and "it returned dummy zero" because this will only happen in cases where there isn't a documented intended meaningful return value.